### PR TITLE
Add keepalives by default to redshift db connection url

### PR DESCRIPTION
**Pull request by austinweisgrau**
These seem to prevent issues with long queries timing out. I couldn't find any great documentation about this, only this stackoverload thread and my own experimentation.

https://stackoverflow.com/q/43594310/18421950

### CODE REVIEW

**Review by ryantimjohn — APPROVED** [review]

**Comment by ryantimjohn** [comment]
@austinweisgrau sorry it took me so long to review but seems like a very reasonable change; record mover is not building currently (see https://circleci.com/gh/bluelabsio/records-mover/16903 ) so we'll have to correct issue with nose-progressive not installing to be able to push this change out to people through Alectrona.

**Comment by jessezlotoff** [comment]
Here's some AWS info for reference: https://docs.aws.amazon.com/redshift/latest/mgmt/connecting-firewall-guidance.html. Pretty sure that the default is 5 mins so anything under that should avoid the issue.

**Comment by archetypalsxe** [comment]
> Seems like @archetypalsxe (Jack) was working on this before he left: #171

:wave: I'm still consulting with BlueLabs! I might be looking back into this soon, there was just a shift of priorities and I ended up working on a bunch of other stuff. The biggest thing preventing us from upgrading to Airflow 2 is the build failing and trying to get that going again

**Comment by ryantimjohn** [comment]
> > Seems like @archetypalsxe (Jack) was working on this before he left: #171
> 
> 👋 I'm still consulting with BlueLabs! I might be looking back into this soon, there was just a shift of priorities and I ended up working on a bunch of other stuff. The biggest thing preventing us from upgrading to Airflow 2 is the build failing and trying to get that going again

Shifting priorities? I don't believe it =p

Is there anything we can help on with the breaking build that you're trying to fix with #171 ? Happy to scrub in if it'd be helpful.

File: tests/unit/db/test_sqlalchemy_driver_picking.py [hunk 1]
```diff
@@ -35,6 +35,8 @@ def test_create_sqlalchemy_url(self,
                 'port': 123,
                 'database': 'analyticsdb'
             }
+            if human_style_db == 'redshift':
+                db_facts['query'] = {'keepalives': '1','keepalives_idle': '30'})
```
Comment (bluelabsbutler) [inline tests/unit/db/test_sqlalchemy_driver_picking.py:39]: E231 missing whitespace after ','

File: tests/unit/db/test_sqlalchemy_driver_picking.py [hunk 2]
```diff
@@ -35,8 +35,11 @@ def test_create_sqlalchemy_url(self,
                 'port': 123,
                 'database': 'analyticsdb'
             }
+            if human_style_db_type in ['redshift','psql (redshift)']:
+                db_facts['query'] = {'keepalives': '1','keepalives_idle': '30'}
```
Comment (bluelabsbutler) [inline tests/unit/db/test_sqlalchemy_driver_picking.py:39]: E231 missing whitespace after ','

File: tests/unit/db/test_sqlalchemy_driver_picking.py [hunk 3]
```diff
@@ -35,8 +35,11 @@ def test_create_sqlalchemy_url(self,
...
             actual_url = connect.create_sqlalchemy_url(db_facts)
-            assert str(actual_url) == expected_url
+            actual_url_str = str(actual_url)
+            self.assertEqual(actual_url_str,expected_url,"{}!={}".format(actual_url_str,expected_url))
```
Comment (bluelabsbutler) [inline tests/unit/db/test_sqlalchemy_driver_picking.py:42]: E231 missing whitespace after ','

Comment (bluelabsbutler) [inline tests/unit/db/test_sqlalchemy_driver_picking.py:42]: E501 line too long (102 > 100 characters)

File: tests/unit/db/test_sqlalchemy_driver_picking.py [hunk 4]
```diff
@@ -71,5 +74,7 @@ def test_create_sqlalchemy_url_odbc_preferred(self,
                 'port': 123,
                 'database': 'analyticsdb'
             }
+            if human_style_db_type == 'redshift':
+                db_facts['query'] = {'keepalives': '1','keepalives_idle': '30'}
```
Comment (bluelabsbutler) [inline tests/unit/db/test_sqlalchemy_driver_picking.py:78]: E231 missing whitespace after ','

**Comment by ryantimjohn** [comment]
All tests passing, going to merge

**Pull request merged**